README.md

```diff
@@ -17,7 +17,65 @@
 
 [letsencrypt]: https://letsencrypt.org/
 
-## Usage
+# Developing with shelf_letsencrypt
+LetsEncrypt provide a few challenges for your development enviroment. 
+Read on for a few hints.
+
+## A word of caution
+LetsEncrypt rate limits the issuing of production certificates.
+It is very easy to get locked out of letsencrypt for an extended period of time (days)
+leaving you in the situation where you can't issue a production certificate.
+
+CRITICAL: you could end up with your production systems down for days!!!!
+
+I would advise you to read up on the Lets Encrypt rate limits:
+
+https://letsencrypt.org/docs/rate-limits/
+
+To avoid this (potentially) major issue make certain that you test with a STAGING
+certificate.
+
+
+You do this by passing in 'production: false' (the default) when creating
+the LetsEncrypt certificate.
+Staging certificates still have rate limits but they are much more generours
+
+```dart 
+final LetsEncrypt letsEncrypt = LetsEncrypt(certificatesHandler, production: false);
+```
+
+
+## Permissions
+On Linux you need to be root (sudo) to open a port below 1024. If you try
+to start your server with the default ports (80, 443) you will fail.
+
```

Comment:
Also add a documentation for `setcap` to avoid execution as root:

Enabling the `dart` VM:
```shell
sudo setcap 'cap_net_bind_service=+ep' /usr/lib/dart/bin/dart
```

Enabling a self-executable (from `dart compile exe your_tool.dart`):
```shell
sudo setcap 'cap_net_bind_service=+ep' your_tool.exe
```